Split out UPGMA and WPGMA

stuckmke opened the pull request:
Updated the TreeConstruction.py script to correctly calculate UPGMA and split out a separate WPGMA method (using what was previously called UPGMA).

Also updated the UPGMA test tree and created a WPGMA test tree (again, using the previous UPGMA code).

This pull request addresses issue #1208 and is a re-do of pull request #1213.  See PR #1213 for full explanation of changes and commentary.  Sorry for the long delay on all of this, I'm new to git and was trying to figure all of this out.

**NOTE:** I believe the test scripts must be updated to account for the new WPGMA method and to utilize the WPGMA test tree.  I'm unsure how to do this, so I appreciate any help in that area.



I hereby agree to dual licence this and any previous contributions under both the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style checks pass with these changes.

I am happy be thanked by name in the ``NEWS.rst`` and ``CONTRIB.rst`` files, if this contribution is deemed significant enough.

peterjc commented on Bio/Phylo/TreeConstruction.py:
```diff
@@ -758,7 +838,11 @@ def _height_of(self, clade):
+                path = clade.get_path(target=terminal)
+                height = 0
+                for value in path:
+                    height = height + value.branch_length
```
Style wise, that could be:

```python
height += value.branch_length
```

(The same suggestion could have been made to the original code too)

stuckmke replied:
Thanks for catching this!  Fixed with 670f2ae

peterjc commented:
@lijax, @JoaoRodrigues - Is this OK to merge? This seems to implement Joao's suggestion and now has tests.

Does it deserve a line in the ``NEWS.rst`` file for the next release?

Also, @stuckmke can you add yourself to the ``NEWS.rst`` and ``CONTRIB.rst`` files (note alphabetical sorting).

stuckmke commented:
Okay, I added my info to `CONTRIB.rst` and `NEWS.rst`.  In the news file, I added my info at the bottom of the top-most Biopython 1.71 section.  Let me know if I need to move anything around.  Thanks!
